journal/src/main/java/io/zeebe/journal/file/record/JournalRecordReaderUtil.java

```diff
@@ -67,9 +64,9 @@ public JournalRecord read(final ByteBuffer buffer, final long expectedIndex) {
           checksumGenerator.compute(buffer, startPosition + metadataLength, recordLength);
 
       if (checksum != metadata.checksum()) {
-        // TODO: Throw an exception, when we introduce magic headers before each record
         buffer.reset();
-        return null;
+        throw new CorruptedLogException(
+            "Record doesn't match checksum. Log must be deleted after shutdown");
```

Comment:
```suggestion
            "Record doesn't match checksum. The log segment may be corrupted.");
```

Reply:
The log message should be improved by giving proper instruction on how to recover. Just deleting the log is not sufficient. But that can be done as part of the other issue.